pkg/controller/seed-controller-manager/mla/mla.go

```diff
@@ -102,20 +107,40 @@ func Add(
 	httpClient := &http.Client{Timeout: 15 * time.Second}
 	grafanaClient := grafanasdk.NewClient(grafanaURL, grafanaAuth, httpClient)
 
-	if err := newOrgGrafanaReconciler(mgr, log, numWorkers, workerName, versions, grafanaClient, mlaEnabled); err != nil {
-		return fmt.Errorf("failed to create mla project controller: %w", err)
-	}
-	if err := newOrgUserGrafanaReconciler(mgr, log, numWorkers, workerName, versions, grafanaClient, mlaEnabled); err != nil {
-		return fmt.Errorf("failed to create mla userprojectbinding controller: %w", err)
-	}
-	if err := newDatasourceGrafanaReconciler(mgr, log, numWorkers, workerName, versions, httpClient, grafanaURL, grafanaAuth, mlaNamespace, overwriteRegistry, mlaEnabled); err != nil {
-		return fmt.Errorf("failed to create mla cluster controller: %w", err)
-	}
-	if err := newAlertmanagerReconciler(mgr, log, numWorkers, workerName, versions, httpClient, cortexAlertmanagerURL, mlaEnabled); err != nil {
-		return fmt.Errorf("failed to create mla alertmanager configuration controller: %w", err)
+	orgGrafanaController := newOrgGrafanaController(mgr.GetClient(), log, grafanaClient)
+	alertmanagerController := newAlertmanagerController(mgr.GetClient(), log, httpClient, cortexAlertmanagerURL)
+	orgUserGrafanaController := newOrgUserGrafanaController(mgr.GetClient(), log, grafanaClient)
+	datasourceGrafanaController := newDatasourceGrafanaController(mgr.GetClient(), httpClient, grafanaURL, grafanaAuth, mlaNamespace, log, overwriteRegistry)
+	userGrafanaController := newUserGrafanaController(mgr.GetClient(), log, grafanaClient, httpClient, grafanaURL, grafanaHeader)
+	if mlaEnabled {
+		if err := newOrgGrafanaReconciler(mgr, log, numWorkers, workerName, versions, orgGrafanaController); err != nil {
+			return fmt.Errorf("failed to create mla project controller: %w", err)
+		}
+		if err := newOrgUserGrafanaReconciler(mgr, log, numWorkers, workerName, versions, orgUserGrafanaController); err != nil {
+			return fmt.Errorf("failed to create mla userprojectbinding controller: %w", err)
+		}
+		if err := newDatasourceGrafanaReconciler(mgr, log, numWorkers, workerName, versions, datasourceGrafanaController); err != nil {
+			return fmt.Errorf("failed to create mla cluster controller: %w", err)
+		}
+		if err := newAlertmanagerReconciler(mgr, log, numWorkers, workerName, versions, alertmanagerController); err != nil {
+			return fmt.Errorf("failed to create mla alertmanager configuration controller: %w", err)
+		}
+		if err := newUserGrafanaReconciler(mgr, log, numWorkers, workerName, versions, userGrafanaController); err != nil {
+			return fmt.Errorf("failed to create mla user controller: %w", err)
+		}
 	}
-	if err := newUserGrafanaReconciler(mgr, log, numWorkers, workerName, versions, grafanaClient, httpClient, grafanaURL, grafanaHeader, mlaEnabled); err != nil {
-		return fmt.Errorf("failed to create mla user controller: %w", err)
+	cleanupController := newCleanupController(
```

Comment:
This means if `mlaEnabled`, we will still run `cleanupController`, what is the reason for that? Will it be flighting with other mla controllers?

Reply:
it won't be fighting cause there is a check for MLA flag in `cleanupController`, but I guess you are right, we can just not run this controller when MLA enabled

Reply:
done